homeassistant/components/climate/venstar.py

```diff
@@ -0,0 +1,287 @@
+        """Return the current temperature."""
+        return self._client.get_indoor_temp()
+
+    @property
+    def current_humidity(self):
+        """Return the current humidity."""
+        return self._client.get_indoor_humidity()
+
+    @property
+    def current_operation(self):
+        """Return current operation ie. heat, cool, idle."""
+        if self._client.mode == self._client.MODE_HEAT:
+            return STATE_HEAT
+        elif self._client.mode == self._client.MODE_COOL:
+            return STATE_COOL
+        elif self._client.mode == self._client.MODE_AUTO:
+            return STATE_AUTO
+        else:
+            return STATE_IDLE
+
+    @property
+    def current_fan_mode(self):
+        """Return the fan setting."""
+        if self._client.fan == self._client.FAN_AUTO:
+            return STATE_AUTO
+        else:
+            return STATE_ON
+
+    @property
+    def state_attributes(self):
+        """Extend the pre-packaged state attributes for extra info."""
+        orig_data = super(VenstarThermostat, self).state_attributes
+        orig_data['fan_state'] = self._client.fanstate
+        orig_data['hvac_state'] = self._client.state
+        return orig_data
+
+# Target Values
+    @property
+    def target_temperature(self):
+        """Return the target temperature we try to reach."""
+        if self._client.mode == self._client.MODE_HEAT:
+            return self._client.heattemp
+        elif self._client.mode == self._client.MODE_COOL:
+            return self._client.cooltemp
+        else:
+            return None
+
+    @property
+    def target_temperature_low(self):
+        """Return the lower bound temp if auto mode is on."""
+        if self._client.mode == self._client.MODE_AUTO:
+            return self._client.heattemp
+        else:
+            return None
+
+    @property
+    def target_temperature_high(self):
+        """Return the upper bound temp if auto mode is on."""
+        if self._client.mode == self._client.MODE_AUTO:
+            return self._client.cooltemp
+        else:
+            return None
+
+    @property
+    def target_humidity(self):
+        """Return the humidity we try to reach."""
+        return self._client.hum_setpoint
+
+    @property
+    def min_humidity(self):
+        """Return the minimum humidity. Hardcoded to 0 in API."""
+        return 0
+
+    @property
+    def max_humidity(self):
+        """Return the maximum humidity. Hardcoded to 60 in API."""
+        return 60
+
+# Commands
+    def set_temperature(self, **kwargs):
+        """Set a new target temperature."""
+        if self._client.mode == self._client.MODE_AUTO:
+            temp_low = kwargs.get(ATTR_TARGET_TEMP_LOW)
+            temp_high = kwargs.get(ATTR_TARGET_TEMP_HIGH)
+        else:
+            temperature = kwargs.get(ATTR_TEMPERATURE)
+
+        if self._client.mode == self._client.MODE_HEAT:
+            _LOGGER.info("Currently operating in heat mode. "
```

Comment:
That is a lot of logging… HA will automatically print a log line when your entity changes its state. So, I'd suggest removing this.